documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -351,4 +353,232 @@ Control characters are not allowed in a specification.
+**Output Message**: Parameter Must have the "name" property defined with non-empty string as its value.
+
+**Description**: A parameter must have a `name` property for the SDK to be properly generated. 
+
+**Why the rule is important**: AutoRest fails to generate code if the `name` property is not provided for a parameter.
+
+**How to fix the violation**: Add a non-empty `name` property to the parameter.
+
+**Bad Example**:
+```
+"MyParam":{
+  "type": "string",
+  "in": "path",
+  "description": "sample param"
+}
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="S1001" />S1001 OperationIdNounVerb
+**Output Message**: Per the Noun_Verb convention for Operation Ids, the noun '{0}' should not appear after the underscore.
+
+**Description**: OperationId should be of the form `Noun_Verb`. 
+
+**Why the rule is important**: AutoRest breaks the operation id into its `Noun` and `Verb` where `Noun` becomes name of the operations class and the `Verb` becomes the name of the method in that class. Not adhering to this format can either cause AutoRest to fail or can generate semantically incorrect SDK.
+
+**How to fix the violation**: Ensure operationId is of the form `Noun_Verb`.
+
+**Bad Example**:
+```
+Activate_Certificate
+CertificateActivate
+```
+**Good Example**:
+```
+Certificate_Activate
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="S2025" />S2025 OneUnderscoreInOperationId
+**Output Message**: Only 1 underscore is permitted in the operation id, following Noun_Verb conventions.
+
+**Description**: An operationId can have exaclty one underscore, not adhering to it can cause errors in code generation.
+
+**Why the rule is important**: Given an operationId of the form `Noun_Verb`, AutoRest breaks the operation id into its `Noun` and `Verb` where `Noun` becomes name of the operations class and the `Verb` becomes the name of the method in that class. Not adhering to this format can cause AutoRest to fail during code generation.
+
+**How to fix the violation**: Ensure operationId is of the form `Noun_Verb` and contains exactly one underscore.
+
+**Bad Example**:
+```
+Activate_Primary_Certificate
+```
+**Good Example**:
+```
+PrimaryCertificate_Activate
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R3023" />R3023 OperationsAPIImplementation
```

Comment:
Table has the name `OperationsAPIImplementationValidation`. Let's make sure which one is correct.